src/orange/univ/SimpleUnitTracker.hh

```diff
@@ -301,8 +305,12 @@ CELER_FUNCTION real_type SimpleUnitTracker::safety(Real3 const& pos,
     if (!vol.simple_safety())
     {
         // Has a tricky surface: we can't use the simple algorithm to calculate
-        // the safety, so return a conservative estimate.
-        return 0;
+        // the safety, use the oriented bounding zone
+        auto const& obz_id
+            = params_.volume_records[unit_record_.volumes[volid]].obz_id;
+        detail::OrientedBoundingZone obz(params_.obz_records[obz_id],
+                                         obz_storage_);
+        return obz.calc_safety_inside(pos);
```

Comment:
Unfortunately it's not going to be this simple: we need the minimum over both "inside current volume" and "outside all directly enclosed volumes". I'm not sure if we have that metadata yet, in fact...